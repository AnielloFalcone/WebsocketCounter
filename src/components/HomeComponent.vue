<template>
    <div id="home">
        <counter-component v-bind:counter="counter"></counter-component>
        <list-component v-if="counter > 0" v-bind:counter="counter"></list-component>
    </div>
</template>

<script>
    import io from "socket.io-client";
    import CounterComponent from "./CounterComponent";
    import ListComponent from "./ListComponent";
    const socket = io.connect("http://localhost:4000", {
        transports: ["polling", "websocket"],
    });

    export default {
        components: {
            CounterComponent,
            ListComponent
        },
        data() {
            return {
                counter: this.getCounter()
            };
        },
        created() {
            window.addEventListener('beforeunload', this.disconnectWS);
            this.registerSocketListeners();
        },
        methods: {
            disconnectWS() {
                socket.disconnect();
            },
            getCounter() {
                this.checkConnection();
                socket.emit('counter:get', counter => {this.setCounter(counter)});
            },
            checkConnection() {
                if (socket.connected === false) {
                    socket.open();
                }
            },
            registerSocketListeners() {
                socket.on('counter:update', counter => {this.setCounter(counter)})
            },
            setCounter(counter) {
                this.counter = counter;
            }
        }
    }
</script>

<style lang="scss" scoped>
    #home {
        display: flex;
        flex-direction: row;
        margin-top: 42px;

        @media screen and (max-width: 576px) {
            flex-direction: column;
            align-items: center;
        }
    }
</style>
