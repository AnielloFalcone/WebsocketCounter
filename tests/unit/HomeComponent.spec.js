import { shallowMount } from "@vue/test-utils";
import HomeComponent from "../../src/components/HomeComponent";

describe("HomeComponent.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(HomeComponent);
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it("renders HomeComponent when passed", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.element).toBeDefined();
    expect(wrapper.element).toMatchSnapshot();
  });
});
