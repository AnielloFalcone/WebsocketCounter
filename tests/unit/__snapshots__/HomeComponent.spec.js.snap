// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent.vue renders HomeComponent when passed 1`] = `
<div
  id="home"
>
  <button
    class="counter-get"
  >
    Add
  </button>
   
  <counter-component-stub />
   
  <!---->
</div>
`;
